log.sh

```diff
@@ -76,9 +76,16 @@ LSLOG () {
   # if no message was passed, read it from STDIN
   local _MSG
   [[ $# -ne 0 ]] && _MSG="$@" || _MSG="$(cat)"
-  echo -ne "$_LS_LEVEL_BEGIN$OUTPUT " >> "$LS_OUTPUT"
-  echo -n  "$_MSG"                    >> "$LS_OUTPUT"
-  echo -e "$_LS_LEVEL_END" >> "$LS_OUTPUT"
+  if [[ "$LS_OUTPUT" = "/dev/stdout" ]]
+  then
```

Comment:
I missed this one, I prefer `then` on the if line:

```
if ... ; then
```

Please do a commit with `--amend` after you change it (don't forget to add first or commit with `-a`), then push with `-f` (force).